module/Finna/src/Finna/Content/Covers/Kirjavalitys.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * Kirjavalitys Cover Image Service cover content loader.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2018.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Kalle Pyykkönen <kalle.pyykkonen@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace Finna\Content\Covers;
+
+use VuFindCode\ISBN;
+
+/**
+ * Kirjavalitys Cover Image Service cover content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Kalle Pyykkönen <kalle.pyykkonen@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Kirjavalitys extends \VuFind\Content\AbstractCover
+{
+    /**
+     * Recordloader to fetch the current record
+     *
+     * @var VuFind\RecordLoader
+     */
+    protected $recordLoader = null;
+
+    /**
+     * Constructor
+     *
+     * @param VuFind\RecordLoader $recordLoader Record loader.
+     */
+    public function __construct(\VuFind\Record\Loader $recordLoader)
+    {
+        $this->recordLoader = $recordLoader;
+        $this->supportsRecordid = true;
+        $this->cacheAllowed = false;
+    }
+
+    /**
+     * Get image URL for a particular API key and set of IDs (or false if invalid).
+     *
+     * @param string $key  Library ID
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        $sizeCodes = [
+            'medium' => 'max400',
+            'small' => 'max200',
+            'large' => 'max800',
+        ];
+        try {
+            $pid = '';
+            $driver = $this->recordLoader->load($ids['recordid'], 'Solr');
+            $recordISBN = new ISBN($driver->getCleanISBN());
+            if ($isbn = $recordISBN->get13()) {
+                $pid = $isbn;
+            } else {
```

Comment:
Voisko muokata muotoon if (!($pid = $recordISBN->get13())) {jäis else pois.

Reply:
Katoin kanssa, että get13 voi palauttaa falsen niin pitäiskö alhaallakin katsoa !$pid eikä verrata === ''

Reply:
Jos sen muuttaa, niin siitä seuraa juurikin tuo, että pitää false käsitellä erikseen. Nykyiselläänhän $isbn asetetaan $pid:iin vain, jos se ei ole falsy. Lisäksi se on hyvin pitkälle samanlainen kuin BTJ:ssä.

Reply:
Ah, sepä selkeytti sen sitten.